utils/s2n_random.c

```diff
@@ -542,38 +549,59 @@ static int s2n_rand_init_cb_impl(void)
     return S2N_SUCCESS;
 }
 
+/* Attempt to detect if the libcrypto is OpenSSL.
+ *
+ * Since, quite a few libcrypto (BoringSSL, AWSLC) implementations are ABI
+ * compatible forks of OpenSSL, it is risky to detect OpenSSL by checking for
+ * macro defines. Instead we detect by eliminating other libcrypto.
+ */
+bool s2n_libcrypto_is_likely_openssl()
+{
+    return !s2n_libcrypto_is_boringssl() && !s2n_libcrypto_is_libressl() && !s2n_libcrypto_is_awslc();
+}
```

Comment:
> Does this method belong in this module?

> Maybe we need to just commit to "s2n_libcrypto_is_openssl"

`s2n_libcrypto_is_openssl` is risky because we cant actually be 100% sure it is OpenSSL. Because of the assumptions, I decided to place the method is in this file because I didnt want it to be generally available across the codebase.

Reply:
AI: search of other uses and move to a single global location so that we can find and update it in the future.